ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Ledger.hs

```diff
@@ -203,3 +216,50 @@ instance CardanoHardForkConstraints c
           :* Nil
           )
     hcollapse <$> (hsequence' $ hap np $ Telescope.tip idx)
+
+instance CardanoHardForkConstraints c => SerializeTablesWithHint (LedgerState (HardForkBlock (CardanoEras c))) where
+  encodeTablesWithHint (HardForkLedgerState (HardForkState idx)) (LedgerTables (ValuesMK tbs)) =
+    let
+      np = (Fn $ const $ K $ Codec.CBOR.Encoding.encodeMapLen 0)
+        :* (Fn $ const $ K $ encOne (Proxy @ShelleyEra))
+        :* (Fn $ const $ K $ encOne (Proxy @AllegraEra))
+        :* (Fn $ const $ K $ encOne (Proxy @MaryEra))
+        :* (Fn $ const $ K $ encOne (Proxy @AlonzoEra))
+        :* (Fn $ const $ K $ encOne (Proxy @BabbageEra))
+        :* (Fn $ const $ K $ encOne (Proxy @ConwayEra))
+        :* Nil
```

Comment:
I recall we made sure elsewhere to make this type of NP into a CAF. Should we do this here as well?

Reply:
I added a comment about it. We would need to lift this class into the HF territory or create a companion class. For now we leave it as is.